### DISCUSSION TITLE
Can I use third party Xamarin.Android UI componets (not for Xamarin.Forms) in Uno?

### DISCUSSION THREAD

**Discussion by ivoryguard** [unoplatform/uno#11281] [Q&A] [answered]
Hello.
I am considering Uno platform to develop an Android App.
It is currently for Android only (it can be extended to other platform in future)
I am very familiar with WPF Xaml, I prefer Xaml to Android UI system of Xamarin.Android.

However, I am afraid that There are not enough 3rd party components for Uno.
So, if I can use Xamarin.Android components in Uno with less performance loss, I will adopt Uno at new project.

Best regards.

**Comment by sasakrsmanovic** [comment, marked as answer]
Thanks for considering. What component provider in particular you are thinking of using? What components are most required? We'd love to hear this for future project development. 

Also, between WinUI controls, Windows Community Toolkit controls, the Uno Toolkit and some open source providers (i.e. LiveCharts), as well as Infragistics or Syncfusion beta controls we think you have plenty of options for UI controls.

Also, if you see this session from Daniel from UnoConf  you will see how native 3rd party controls can be used on Android  https://platform.uno/blog/using-native-controls-in-uno-platform-applications-demo-code-included/

**Reply by ivoryguard** [reply]
Thank you for the reply.
I need a full-featured PDF viewer/renderer component, but I cannot find anything at Uno basic components and other 3rd party components.

But, there is a few PDF viewer/render component for Xamarin.Android and Android Native.

So, If I can use it through a renderer like Xamarin.Forms, I will get much more component in available at Uno platform.

Best regards.

**Reply by sasakrsmanovic** [reply]
Yes you are. Please see this https://platform.uno/docs/articles/native-views.html
